src/AbstractUri.php

```diff
@@ -384,25 +384,28 @@ private function formatIp(string $host)
         }
 
         static $ip_future = '/^
-            v(?<version>\d+)\.
-            ((?<unreserved>[a-z0-9_~\-\.])|(?<sub_delims>[!$&\'()*+,;=:]))+
+            v(?<version>[A-F0-9])+\.
+            (
```

Comment:
This should be a non-matching group, so `(?:` instead of `(`.

Reply:
good catch will change that